testData/parsing/haxe/expressions/ComplexExpression.txt

```diff
@@ -0,0 +1,157 @@
+              TYPE_TAG
+                HaxePsiToken::(':')
+                TYPE_OR_ANONYMOUS
+                  TYPE
+                    REFERENCE_EXPRESSION
+                      IDENTIFIER
+                        HaxePsiToken:ID('Int')
+              VAR_INIT
+                HaxePsiToken:=('=')
+                LITERAL_EXPRESSION
+                  HaxePsiToken:LITINT('4')
+            HaxePsiToken:;(';')
+          LOCAL_VAR_DECLARATION
+            HaxePsiToken:var('var')
+            LOCAL_VAR_DECLARATION_PART
+              COMPONENT_NAME
+                IDENTIFIER
+                  HaxePsiToken:ID('i3')
+              TYPE_TAG
+                HaxePsiToken::(':')
+                TYPE_OR_ANONYMOUS
+                  TYPE
+                    REFERENCE_EXPRESSION
+                      IDENTIFIER
+                        HaxePsiToken:ID('Int')
+              VAR_INIT
+                HaxePsiToken:=('=')
+                LITERAL_EXPRESSION
+                  HaxePsiToken:LITINT('2')
+            HaxePsiToken:;(';')
+          ASSIGN_EXPRESSION
+            REFERENCE_EXPRESSION
+              IDENTIFIER
+                HaxePsiToken:ID('i1')
+            ASSIGN_OPERATION
+              HaxePsiToken:=('=')
+            ASSIGN_EXPRESSION
+              REFERENCE_EXPRESSION
+                IDENTIFIER
+                  HaxePsiToken:ID('i3')
+              ASSIGN_OPERATION
+                HaxePsiToken:=('=')
+              ASSIGN_EXPRESSION
+                REFERENCE_EXPRESSION
+                  IDENTIFIER
+                    HaxePsiToken:ID('i2')
+                ASSIGN_OPERATION
+                  SHIFT_ASSIGN_OPERATOR
+                    UNSIGNED_SHIFT_RIGHT_ASSIGN_OPERATOR
+                      HaxePsiToken:>>>=('>>>=')
+                LITERAL_EXPRESSION
+                  HaxePsiToken:LITINT('1')
+          HaxePsiToken:;(';')
+          ASSIGN_EXPRESSION
+            REFERENCE_EXPRESSION
+              IDENTIFIER
+                HaxePsiToken:ID('i1')
+            ASSIGN_OPERATION
+              SHIFT_ASSIGN_OPERATOR
+                SHIFT_RIGHT_ASSIGN_OPERATOR
+                  HaxePsiToken:>>=('>>=')
+            ASSIGN_EXPRESSION
+              REFERENCE_EXPRESSION
+                IDENTIFIER
+                  HaxePsiToken:ID('i2')
+              ASSIGN_OPERATION
+                SHIFT_ASSIGN_OPERATOR
+                  SHIFT_RIGHT_ASSIGN_OPERATOR
+                    HaxePsiToken:>>=('>>=')
+              LITERAL_EXPRESSION
+                HaxePsiToken:LITINT('1')
+          HaxePsiToken:;(';')
+          ASSIGN_EXPRESSION
+            REFERENCE_EXPRESSION
+              IDENTIFIER
+                HaxePsiToken:ID('i1')
+            ASSIGN_OPERATION
+              HaxePsiToken:+=('+=')
+            ASSIGN_EXPRESSION
+              REFERENCE_EXPRESSION
+                IDENTIFIER
+                  HaxePsiToken:ID('i3')
+              ASSIGN_OPERATION
+                SHIFT_ASSIGN_OPERATOR
```

Comment:
Hm, I think it's better to leave them as is, otherwise it requires to make some another changes with code, I think I'll find better solution during semantic reworking in next pull requests